client/app/pages/ItemsList.jsx

```diff
@@ -0,0 +1,325 @@
+    const request = {
+      page: requestedPage,
+      page_size: itemsPerPage,
+      order: orderByField,
+      tags: this.state.selectedTags,
+    };
+    if (this.isInSearchMode) {
+      request.q = this.state.searchTerm;
+    }
+    return request;
+  }
+
+  toggleSorting(field) {
+    this.state.paginator.orderBy(field);
+  }
+
+  // eslint-disable-next-line no-unused-vars, class-methods-use-this
+  doRequest(request) {
+    return null;
+  }
+
+  processResponse() {
+    this.setState({ isLoaded: true });
+  }
+
+  update(state) {
+    this.setState(state, () => {
+      this.updatePaginator();
+    });
+  }
+
+  updatePaginator(updates) {
+    const paginator = this.state.paginator;
+    updates = extend({ page: paginator.page }, updates);
+    paginator.setPage(
+      updates.page,
+      updates.pageSize,
+      updates.pageOrder,
+      updates.pageOrderReverse,
+    );
+    // this.forceUpdate();
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  renderPageHeader() {
+    return (
+      <PageHeader title={$route.current.title} />
+    );
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  renderLoadingState() {
+    return (
+      <div className="text-center">
+        <BigMessage icon="fa-spinner fa-2x fa-pulse" message="Loading..." />
+      </div>
+    );
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  renderEmptyState() {
+    return (
+      <div className="text-center">
+        <BigMessage icon="fa-search" message="Sorry, we couldn't find anything." />
+      </div>
+    );
+  }
+
+  renderSearchInput(placeholder = null) {
+    return (
+      <div className="m-b-10">
+        <Input
+          className="form-control"
+          placeholder={placeholder}
+          defaultValue={this.state.searchTerm}
+          onChange={event => this.update({ searchTerm: event.target.value })}
+          autoFocus
+        />
+      </div>
+    );
+  }
+
+  renderSidebarMenu() {
+    const sidebarMenu = this.getSidebarMenu();
+    if (sidebarMenu.length > 0) {
+      return (
+        <div className="list-group m-b-10 tags-list tiled">
+          {map(sidebarMenu, item => (
+            <a
+              key={item.key}
+              href={item.href}
+              className={'list-group-item ' + (this.props.currentPage === item.key ? 'active' : '')}
+            >
+              {isString(item.icon) && <span className="btn-favourite m-r-5"><i className={item.icon} aria-hidden="true" /></span>}
+              {isFunction(item.icon) && (item.icon(item) || null)}
+              {item.title}
+            </a>
+          ))}
+        </div>
+      );
+    }
+    return null;
+  }
+
+  renderTagsList(tagsUrl = null) {
+    if (isString(tagsUrl) && (tagsUrl !== '')) {
+      return (
+        <div className="m-b-10">
+          <TagsList tagsUrl={tagsUrl} onUpdate={selectedTags => this.update({ selectedTags })} />
+        </div>
+      );
+    }
+    return null;
+  }
+
+  renderPageSizeSelect() {
```

Comment:
How about we remove this and use Pagination's built in support for showing page size? Just need to add the following options:

```
showSizeChanger onShowSizeChange={} pageSize={} pageSizeOptions={}
```

(note that their `pageSizeOptions` takes a list of strings...)

Reply:
@arikfr Do you mean this one?
![image](https://user-images.githubusercontent.com/12139186/52115547-8d67bc80-2617-11e9-8524-793fddd565aa.png)
The only issue is that currently we hide entire component when there is the only page. Now we will need to show it, otherwise page size selector will be hidden too (example: 10 items total, page size was 5 = 2 pages; user changed it to 20 = 1 page - if pagination is hidden - no way to change page size).

Reply:
#3386.